crates/store/src/db/mod.rs

```diff
@@ -108,8 +116,26 @@ impl NoteRecord {
         let execution_hint = column_value_as_u64(row, 8)?;
         let merkle_path_data = row.get_ref(9)?.as_blob()?;
         let merkle_path = MerklePath::read_from_bytes(merkle_path_data)?;
-        let details_data = row.get_ref(10)?.as_blob_or_null()?;
-        let details = details_data.map(<Vec<u8>>::read_from_bytes).transpose()?;
+
+        let assets = row.get_ref(10)?.as_blob_or_null()?;
+        let inputs = row.get_ref(11)?.as_blob_or_null()?;
+        let script = row.get_ref(12)?.as_blob_or_null()?;
+        let serial_num = row.get_ref(13)?.as_blob_or_null()?;
+
+        let details = if let (Some(assets), Some(inputs), Some(script), Some(serial_num)) =
```

Comment:
Maybe unnecessary but I wonder if there should be a `debug_assert` here that either all of these are `Some` or all of these are `None`